### PR TITLE
Alter URIFromObjectReference to return apierrs

### PR DESCRIPTION
**Pull request by lberk**
This allows for reconcile deletion to properly remove finalizer and
delete sinkbinding

Followup from #1647

### CODE REVIEW

File: resolver/addressable_resolver.go [hunk 1]
```diff
@@ -137,16 +138,17 @@ func (r *URIResolver) URIFromKReference(ref *duckv1.KReference, parent interface
 // URIFromObjectReference resolves an ObjectReference to a URI string.
 func (r *URIResolver) URIFromObjectReference(ref *corev1.ObjectReference, parent interface{}) (*apis.URL, error) {
 	if ref == nil {
-		return nil, errors.New("ref is nil")
+		return nil, apierrs.NewBadRequest(fmt.Sprintf("ref is nil"))
```
Comment (vagababov) [inline resolver/addressable_resolver.go:141]:
```suggestion
		return nil, apierrs.NewBadRequest("ref is nil")
```

Comment (lberk) [reply resolver/addressable_resolver.go]: ah thanks, fixed

**Review by vagababov — COMMENTED** [review]
/lgtm

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Comment by matzew** [comment]
@mattmoor mind taking a look here ?

**Comment by mattmoor** [comment]
> @mattmoor mind taking a look here ?

I'd 302'd to
/assign @vaikas @n3wscott 
via chat yesterday.  I also told @lberk this
/lgtm

🤔 Feels like haiku 😉

**Review by n3wscott — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**